docs/_articles/en/v3.x/salesforce-graph-engine/rules.md

```diff
@@ -243,46 +237,120 @@ These common scenarios trigger a violation from MultipleMassSchemaLookupRule.
+
+#### UnimplementedTypeRule Limitations
+
+Because UnimplementedType rule excludes `global` scoped classes from consideration, these classes are prevented from being thrown as false positives and aren’t false negatives.
+
+## Pilot Rules
 
-**Example**: 
-```sfdx scanner:run --engine sfge --projectdir /project/dir --target /project/dir/target1```
+### AvoidDatabaseOperationInLoop
 
-UnusedMethodRule recognizes these entry points.
+AvoidDatabaseOperationInLoop detects database operations that occur inside loops and which cause performance degradation. Database operations within loops cause performance degradations and exceed Salesforce Governor Limits. 
+
+To remediate database operation violations, move these operations outside of loops. Follow these best practices.
+
+* Instead of querying objects within a loop, use a SOQL For Loop to iterate over a query’s results..
+
+	For example, instead of:
+	```for (String name : names) {
+		Account[] accs = [SELECT Name FROM Account WHERE Name =: name];```
+	Use:
+	```for (Account[] accs : [SELECT Name FROM Account WHERE Name IN :names]) {```
+
+* Instead of putting database operations within a loop, use the loop to iteratively construct lists of objects, and then perform the database operation on the full list after the loop.
+  
+	For example, instead of:
+	```for (String n : names) {
+  	  insert new Account(Name = n);```
+	Use:
+	```Account[] accs = new List<Account>();
+	for (String n : names) {
+   	 accs.add(new Account(Name = n));
+	}
+	insert accs;```
+
+#### Definition
+
+| Rule Component | Definition                                  																		|
+| ---------		 | ---------                                															  			|
+|**Source**		 |																													|
+|		 	  	 | `@AuraEnabled`-annotated methods     																			|
+|				 |`@InvocableMethod`-annotated methods																			  	|
+|     			 | `@NamespaceAccessible`-annotated methods 																		|
+| 				 |`@RemoteAction`-annotated methods																				  	|
+|				 |Any method returning a `PageReference` object																	  	|
+|				 |`public`-scoped methods on Visualforce Controllers																|
+|				 |`global`-scoped methods on any class																			  	|
+|				 |`Messaging.InboundEmailResult handleInboundEmail()` methods on implementations of `Messaging.InboundEmailHandler`	|
+|				 |Any method targeted during invocation																				|
+| **Sink**		 | 																													|
+| 		  	  	 |Any database operation													|						
+
+#### Interpreting AvoidDatabaseOperationInLoop Results
+
+*Loop Case* 
+
+> A database operation occurred inside a loop.` [%s at %s:%d]`
+
+Explanation:
+
+Your code executes a database operation inside a loop statement. Modify your code to move the database operation outside the loop, then rescan your code.
+
+### RemoveUnusedMethod Rule
```

Comment:
Trailing "Rule" is unneeded. Also, add an anchor:
`### RemoveUnusedMethod <a name='RemoveUnusedMethod'>#</a>`

Reply:
@jfeingold35 I added Rule to the two rules that don't have "rule" in their names, on purpose. Why don't you want Rule in the title?
Anchors added throughout.